web/styles/zulip.css

```diff
@@ -1128,11 +1128,13 @@ td.pointer {
     animation: rotate 1s infinite linear;
 }
 
-/* The way this overrides the menus with a background-color and a high
-   z-index is kinda hacky, and requires some annoying color-matching,
-   but it works. */
 .alert-msg {
     color: hsl(170deg 48% 54%);
+    /* Duplicate the message-content background color so as to hide
+       the hover controls and time marker. */
+    background-color: var(--color-background-stream-message-content);
+    /* The z-index is necessary to ensure that alert messages fully
+       overlay the hover controls and time marker. */
```

Comment:
This also needs to handle `mentioned` messages which have a different background color, which makes me think maybe playing with `visibility` of controls and time could be a better path.

Reply:
Ah, that's a really good catch, Aman. Let me think on this.